src/runtime/tests/schema-test.ts

```diff
@@ -443,4 +444,86 @@ describe('schema', () => {
     assert.strictEqual(schema.fields.age.type, 'Number');
     assert.strictEqual(schema.fields.custom.type, 'Bytes');
   });
+
+
+  const partialOrderGreaterThan = (testCase) => {
+    const [caseName, first, second] = testCase;
+    it(`partially orders entities: ${caseName}`, () => {
+      assert.isTrue(first.isMoreSpecificThan(second));
+      assert.isFalse(second.isMoreSpecificThan(first));
+    });
+  };
+
+  const notPartialOrdered = (testCase) => {
+    const [caseName, first, second] = testCase;
+    it(`determines entities have no overlap: ${caseName}`, () => {
+      assert.isFalse(first.isMoreSpecificThan(second));
+      assert.isFalse(second.isMoreSpecificThan(first));
+    });
+  };
+
+  const partiallyOrderedTestCases = async () => {
+    const manifest = await Manifest.parse(`
+    particle Foo
+      inout Data Thing {Text t, Number n} input1
+      in Object Data Thing {Text t, Number n, Bytes b} input2
+      out [* {Number n, Bytes b}] output1
+      out [Object {Number n, Bytes b}] output2`);
+
+    const particle = manifest.particles[0];
+    const schema = (conn: HandleConnectionSpec): Schema => conn.type.getEntitySchema();
+    const input1 = schema(particle.connections[0]);
+    const input2 = schema(particle.connections[1]);
+    const output1 = schema(particle.connections[2]);
+    const output2 = schema(particle.connections[3]);
+
+    [
+      ['input2  > input1   | in and inout type', input2, input1],
+      ['input2  > output2  | in and out type', input2, output2],
+      ['input2  > output2  | anonymous type', input2, output1],
+      ['output2  > output1 | same type, anonymous is smaller', output2, output1]
+    ]
+      .forEach(partialOrderGreaterThan);
+
+    [
+      ['input1 /> output2 | adjacent types', input1, output2],
+      ['input1 /> output1 | adjacent types, one anonymous', input1, output1],
+    ]
+      .forEach(notPartialOrdered);
+  };
+
+  void partiallyOrderedTestCases();
+
+  it('represent anonymous schema names in a comparable way', async () => {
+    const manifest = await Manifest.parse(`
+    particle Foo
+      in * {Text t, Number n} input1
+      in [* {Number n, Bytes b, Text t}] input2
+      out [* {Number n, Bytes b}] output1
+      out * {Number n, Bytes b} output2`);
+
+    const particle = manifest.particles[0];
+    const schema = (conn: HandleConnectionSpec): Schema => conn.type.getEntitySchema();
+    particle.connections
+      .map(schema)
+      .forEach((s: Schema) => assert.deepEqual(s.names, []));
+  });
+
+  function testPoorAnonConstruction() {
```

Comment:
This test is complex and I have to warm up my brain to understand it - that is not good. Tests should be trivial if possible.

What you want, once you fix the problem at the root is something like this:

```
  it('parses annonymous schemas', async () => {
    const manifest = await Manifest.parse(`alias schema * as X`);
    const schema = manifest.findSchemaByName('X');

    assert.isEmpty(schema.names);
    assert.isEmpty(schema.fields);
  });

  it('parses annonymous inline schemas', async () => {
    const manifest = await Manifest.parse(`
      particle P
        in * {} thing`);

    const particle = manifest.particles[0];
    const connection = particle.handleConnections[0];
    const schema = connection.type.getEntitySchema();

    assert.isEmpty(schema.names);
    assert.isEmpty(schema.fields);
  });
```

Sorry for doing your job for you, but sometimes it is easier to show with an example rather than talk abstractly.

Do you see how simple these tests are? There is no iteration over collection to then programmatically generate inputs and tests cases using them. This is what you should strive for - code that is as simple and as readable and as focused as you can make it.